control-plane/cni/main.go

```diff
@@ -135,56 +162,93 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 	client, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
-		return fmt.Errorf("Error initializing Kubernetes client: %s", err)
+		return fmt.Errorf("error initializing Kubernetes client: %s", err)
 	}
 
 	pod, err := client.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("error retrieving pod: %s", err)
 	}
 
-	// TODO: Add tests for all of the logic below
-	// TODO: Remove has hasBeenInjected and instead look for consul.hashicorp.com/cni-proxy-config annotation
-	// TODO: Add wait and timeout for annotations to show up
-	if hasBeenInjected(*pod) {
-		// If everything is good, add an annotation to the pod
-		// TODO: Remove this as it is just a stub to prove that we can do kubernetes things with the plugin
-		annotations := map[string]string{
-			keyCNIStatus: "true",
-		}
-		pod.SetAnnotations(annotations)
-		_, err = client.CoreV1().Pods(podNamespace).Update(context.TODO(), pod, metav1.UpdateOptions{})
-		if err != nil {
-			return err
-		}
-	} else {
-		logger.Debug("skipping traffic redirect on un-injected pod")
+	// Skip traffic redirection the correct annotations are not on the pod.
+	if skipTrafficRedirection(*pod, retries) {
+		logger.Debug("skipping traffic redirect on un-injected pod: %s", pod.Name)
+		return types.PrintResult(result, cfg.CNIVersion)
+	}
+
+	pod.Annotations[keyTransparentProxyStatus] = waiting
```

Comment:
This is a simple way to show some sort of status that the CNI plugin is working. I have added a simple `updateTransparentProxyStatusAnnotation` function to clean it up a bit so that we can have a `waiting` and `complete` annotation.